backend/services/battles.py

```diff
@@ -36,30 +34,31 @@ def battle_round(creator_pokemon, opponent_pokemon):
 
 def get_battle_winner(battle):
     battle_score = {"creator": 0, "opponent": 0}
-    battle_info = Battle.objects.filter(id=battle.id).values()[0]
 
-    # get the pokemons ids from the battles
-    creator_pokemons_id = [battle_info["creator_pokemon_" + str(i) + "_id"] for i in range(1, 4)]
-    opponent_pokemons_id = [battle_info["opponent_pokemon_" + str(i) + "_id"] for i in range(1, 4)]
-    # get the pokemons from the DB
-    creator_pokemon_list = [Pokemon.objects.filter(id=j).values()[0] for j in creator_pokemons_id]
-    opponent_pokemon_list = [Pokemon.objects.filter(id=j).values()[0] for j in opponent_pokemons_id]
+    # get the batween the battle and the trainers
+    creator = Team.objects.filter(battle=battle, trainer=battle.creator.id)
+    opponent = Team.objects.filter(battle=battle, trainer=battle.opponent.id)
 
-    for creator_pokemon, opponent_pokemon in zip(creator_pokemon_list, opponent_pokemon_list):
+    # get the relation between the trainer's team and its pokemons
+    creator_pokemon_list = PokemonTeam.objects.filter(team=creator[0])
+    opponent_pokemon_list = PokemonTeam.objects.filter(team=opponent[0])
+
+    # get the pokemons names from each instance
+    creator_team = [pokemon.pokemon for pokemon in creator_pokemon_list]
```

Comment:
If you only need the pokemon names, you can do in a single line: `creator_team = PokemonTeam.objects.filter(team=creator).values_list('pokemon', flat=True)`

Reply:
It doesn't work, because it returns their ids instead of their names. I'm not sure I understand this line of code. What does the "flat=True" do?

Reply:
Please forget my suggestion, you need the whole pokemon objects, not their names or ids.

If you wanted only names, you could do: 

```python
PokemonTeam.objects.filter(team=creator).values_list('pokemon__name', flat=True)
```

This code above will get the names as a list.
About `flat=True`, it is harder to explain, nevermind it now. Save this to check later: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#values

Reply:
Thing is: [pokemon.pokemon for pokemon in creator_pokemon_list] doesn't return just the name. It returns the object <Pokemon: pidgeot>, where I can get the attributes. If I use the values_list, I get a list with the pokemons names, which gives the error "name as no attribute .attack"

Reply:
That's right, I was trying to explain the values_list, but I understand my first suggestion was wrong. No worries.